text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,502 @@
+
+### What is not supported?
+* Service Managed Stack Sets
+* Stack Set Instance Add/Remove through CDK
+* Stack Set deletion
+* Docker Assets
+
+## Internal FAQ
+### Why are we doing this?
+CDK does not support the creation and updating of CFN stack sets directly, only through an intermediate single stack. This would also enable customers with existing stack sets to utlize CDK to update in the future.
+
+### Why should we _not_ do this?
+CDK does provide the ```cdk.CFNStackSet``` construct, but requires an intermediate single stack to deploy the stack set which does not monitor stack set operations on update.
+
+### What changes are required to enable this change?
+Changes are required since CDK currently does not have the concept/logic of deploying stack sets directly.
+
+### Is this a breaking change?
+No - this change does not require customers with existing apps to update code to consume the new CDK version.
+
+### What are the drawbacks of this solution?
+The handling of file assets is more complex than a normal stack CDK app since stack set instances are deployed independently from CDK context and exist in different accounts and regions. This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+An alternative is the usage of ```cdk.CFNStackSet```, which is not ideal for customers with existing stack sets or need to monitor stack set operations during updates.
+
+### What is the high level implementation plan?
+At a high level, the implementation will be done in two phases:
+* Bootstrapless StackSet
+  * Create StackSet core construct
+  * Create StackSet artifact type for manifest
+  * Create Bootstrapless StackSet Synthesizer
+  * Update CLI to collect StackSet artifact types in addition to Stacks
+* StackSet with file asset support
+  * Create Process for bootstrapping to enable file asset support
+  * Create StackSet Synthesizer for stack sets that inclue file assets
+
+### Are there any open issues that need to be addressed later?
+* Docker asset management
+* Future support for service managed stack sets
+
+---
+# Implemenation Details
+
+# Requirements
+## In Scope Features
+* Create Self-Managed Stack Set
+  * This should create the stack set directly, as opposed to the [CfnStackSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudformation.CfnStackSet.html) construct which creates a CFN stack that deploys a stack set
+* Update Self-Managed Stack Set
+  * Update includes updating **all existing** stack set instances 
+  * Set Operation Preference Configuration
+* File Assets Management
+
+## Out of Scope Features
+* Delete Self-Managed Stack Set
+* Service Managed Stack Set
+* Add/Remove stack set instances through CDK
+* Update to only specific stack set instances
+* Stack Set Instance Level Parameter Overrides
+
+## Assumptions
+* Management Account contains Stack Set administration role and child accounts contains Stack Set execution role for enabling stack set operations
```

Comment:
Not sure I understand this assumption... this should be done as part of `cdk bootstrap` of the child accounts, right?

If there are changes we need to make to `cdk bootstrap` to support this, they should be specified in this RFC.

Reply:
Yeah, so I'm on the fence. This is a resource that "belongs" to cloud formation stack set, vs something that is specific to cdk. Customer may already have the stack set administration role created, or could create through a stack in CDK.

As for the stack set execution role, this exists in the child account, which CDK would not have context about

Reply:
Sorry,  I still need more details here.

- How is the management account administration Role created? Manually? Through the CDK? How do we set in the CDK what Role is used?
- How is the execution Role created? Manually? Using CDK?
- Do we require that an account+region be `cdk bootstrap`ped before a StackSet instance of a StackSet managed through the CDK can be deployed to it? Why yes/no?
- How does the CloudFormation service know which Role to use when performing the StackSet instance deployment? Can that be set in the CDK?

All of these questions should be answered here.